kn/fotos/entities.py

```diff
@@ -269,42 +282,26 @@ def update_metadata(self):
             if 'DateTimeOriginal' in exif:
                 self.created = datetime.datetime.strptime(exif['DateTimeOriginal'], '%Y:%m:%d %H:%M:%S')
 
+        if self.size is None:
+            self.size = img.size
+
+        return True
 
     def _cache(self, cache):
         if cache == 'full':
-            return {}
+            return
         source = self.original_path
         target = self.get_cache_path(cache)
         target_dir = os.path.dirname(target)
         if not os.path.exists(target_dir):
             os.makedirs(target_dir)
-        if cache == 'thumb':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '200x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'thumb2x':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '400x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'large':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '850x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'large2x':
-            subprocess.check_call(['convert',
-                             source,
-                             '-quality', '90',
-                             '-resize', '1700x',
-                             '-rotate', str(self.rotation),
-                             target])
-        # No worries: Image.open is lazy and will only read headers
-        return {'size': Image.open(target).size}
+
```

Comment:
Oeps, die had ik gemist.
Is dat echt nodig?
Ik zou geen onderscheid maken tussen `large` en `large2x`. Bovendien is 90 al best hoog voor op het web, 85 is al een aardige kwaliteit, en is het voor `large2x` nog minder nodig om een hogere kwaliteit te hebben aangezien die toch al voor hogere resolutie schermen is.
Als we al een kwaliteit instellen, zou ik dat voor allemaal doen en niet maar voor eentje.
Thumbnails mogen wat mij betreft sowieso een kwaliteit van 85 hebben, die zijn niet om de foto zelf echt te bekijken.